src/block/tile/MonsterSpawner.php

```diff
@@ -130,4 +140,126 @@ protected function addAdditionalSpawnData(CompoundTag $nbt) : void{
 
 		$nbt->setFloat(self::TAG_ENTITY_SCALE, $this->displayEntityScale);
 	}
+
+	public function onUpdate() : bool{
+		if($this->closed || $this->entityTypeId === ":"){
+			return false;
+		}
+		if($this->spawnDelay > 0){
+			$this->spawnDelay--;
+			return true;
+		}
+		$position = $this->getPosition();
+		$world = $position->getWorld();
+		if($world->getNearestEntity($position, $this->requiredPlayerRange, Player::class) === null){
+			return true;
+		}
+		$count = 0;
+		foreach($world->getNearbyEntities(
+			AxisAlignedBB::one()
+				->offset($position->x, $position->y, $position->z)
+				->expand($this->spawnRange * 2 + 1, 8, $this->spawnRange * 2 + 1)
```

Comment:
This is scalable according to the entity

Reply:
I'm not sure of the exact vanilla implementation, I got the spawnRange * 2 + 1 from wiki, how should I make it?

Reply:
Zombie spawner as not the same spawn range than blaze or magma cube. If we want to implement it in a good way, we need an API to define this range